CONTRIBUTING.md

```diff
@@ -51,18 +51,19 @@ Best practices for writing test cases:
   ```
 * Project `camunda-engine`: If you need a process engine with custom configuration, use the JUnit rule `org.camunda.bpm.engine.test.util.ProcessEngineBootstrapRule` and chain it with `org.camunda.bpm.engine.test.util.ProvidedProcessEngineRule` like so:
   ```
-  protected ProcessEngineBootstrapRule bootstrapRule = new ProcessEngineBootstrapRule() {
-    public ProcessEngineConfiguration configureEngine(ProcessEngineConfigurationImpl configuration) {
+  protected ProcessEngineBootstrapRule bootstrapRule = new ProcessEngineBootstrapRule(configuration -> {
       // apply configuration options here
-
-      return configuration;
-    }
-  };
+  });
   protected ProvidedProcessEngineRule engineRule = new ProvidedProcessEngineRule(bootstrapRule);
 
   @Rule
   public RuleChain ruleChain = RuleChain.outerRule(bootstrapRule).around(engineRule);
   ```
+* Some Test classes extend the `org.camunda.bpm.engine.test.util.PluggableProcessEngineTest` class. This is a 
+  compatibility class to ease the transition of Test classes from JUnit3 to JUnit4. If you need to modify the 
+  `ProcessEngineConfiguration` in a class that extends `PluggableProcessEngineTest`, please remove 
+  the `extends PluggableProcessEngineTest` inheritance, and add the necessary JUnit4 Rules and Process Engine Services 
+  directly in the class you are modifying, as described in the point above.
```

Comment:
Do we really want to discourage using `PluggableProcessEngineTest`? It contains some common code that is quite handy to simply reuse (e.g. assigning the services to fields) instead of duplicating it.

Reply:
I wrote that section to explain that, if a custom configuration needs to be applied to the `ProcessEngineBootstrapRule`, this isn't really possible when a class is extending the `PluggableProcessEngineTest` since we can't pass the `bootstrapRule` to the `ProvidedProcessEngineRule` instance.

I can rephrase it so it's clearer. 

If the default configuration needs to be used, it's completely fine to use extend the `PluggableProcessEngineTest`.